pkgs/build-support/rust/carnix.nix

```diff
@@ -0,0 +1,536 @@
+{ pkgs, ... }:
```

Comment:
What is this file for? Give it a header explaining its purpose, as well as its structure. The top part consists of functions, and the there are derivations, right? The other files could use headers as well.

Reply:
Done, thanks.